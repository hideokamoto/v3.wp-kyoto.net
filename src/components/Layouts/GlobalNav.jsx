import React from 'react'

// Semantic UI
import {
  Responsive,
} from 'semantic-ui-react'

// component
import PcGlobalNav from '../Navigations/PC/GlobalNav';
import MobileGlobalNav from '../Navigations/Mobile/GlobalNav';

const GlobalNav = () => {
  return [
    <Responsive as={PcGlobalNav} minWidth={992} key={0} />,
    <Responsive as={MobileGlobalNav} maxWidth={991} key={1} />
  ]
}

export default GlobalNav
