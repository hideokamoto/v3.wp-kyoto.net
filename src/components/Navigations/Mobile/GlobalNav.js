import React from 'react'

// Router
import { Link } from 'react-router-dom'

// Semantic UI
import { Dropdown } from 'semantic-ui-react'

class GlobalNav extends React.Component {
  render () {
    return (
      <Dropdown text="Menu" fluid>
        <Dropdown.Menu>
          <Dropdown.Item>
            <Link to="/">Home</Link>
          </Dropdown.Item>
          <Dropdown.Item>
            <Link to="/products">Products</Link>
          </Dropdown.Item>
          <Dropdown.Item>
            <Link to="/about">About</Link>
          </Dropdown.Item>
          <Dropdown.Item>
            <Link to="/en">English</Link>
          </Dropdown.Item>
          <Dropdown.Item>
            <a
              href="https://profiles.wordpress.org/hideokamoto"
              target="_blank"
              rel="noopener noreferrer"
            >
              WordPress.org
            </a>
          </Dropdown.Item>
          <Dropdown.Item>
            <a
              href="https://github.com/hideokamoto/"
              target="_blank"
              rel="noopener noreferrer"
            >
              GitHub
            </a>
          </Dropdown.Item>
          <Dropdown.Item>Login</Dropdown.Item>
        </Dropdown.Menu>
      </Dropdown>
    )
  }
}

export default GlobalNav
