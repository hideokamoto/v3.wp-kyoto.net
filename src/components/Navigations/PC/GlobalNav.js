import React from 'react'

// Router
import { Link } from 'react-router-dom'

// Semantic UI
import { Menu } from 'semantic-ui-react'

class GlobalNav extends React.Component {
  constructor (props) {
    super(props)
    this.state = { activeItem: 'home' }
    this.handleItemClick = this.handleItemClick.bind(this)
  }
  handleItemClick (e, { name }) {
    this.setState({ activeItem: name })
  }
  render () {
    const { activeItem } = this.state
    return (
      <Menu pointing secondary>
        <Menu.Item
          as={Link}
          to="/"
          name="home"
          active={activeItem === 'home'}
          onClick={this.handleItemClick}
        />
        <Menu.Item
          as={Link}
          to="/products"
          name="products"
          active={activeItem === 'products'}
          onClick={this.handleItemClick}
        />
        <Menu.Item
          as={Link}
          to="/about"
          name="about"
          active={activeItem === 'about'}
          onClick={this.handleItemClick}
        />
        <Menu.Item
          as={Link}
          to="/en"
          name="english"
          active={activeItem === 'english'}
          onClick={this.handleItemClick}
        />
        <Menu.Item
          as="a"
          href="https://github.com/hideokamoto/"
          target="_blank"
          rel="noopener noreferrer"
          name="GitHub"
        />
      </Menu>
    )
  }
}

export default GlobalNav
