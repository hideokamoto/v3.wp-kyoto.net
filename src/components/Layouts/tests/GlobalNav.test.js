/* eslint-disable react/jsx-filename-extension */
import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import GlobalNav from '../GlobalNav'
configure({ adapter: new Adapter() })

describe('GlobalNav', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<GlobalNav />)
    expect(wrapper).toMatchSnapshot()
  })
})
