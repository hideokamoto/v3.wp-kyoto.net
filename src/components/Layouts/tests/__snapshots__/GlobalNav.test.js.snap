// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalNav should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GlobalNav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <Responsive
      as={[Function]}
      minWidth={992}
    />,
    <Responsive
      as={[Function]}
      maxWidth={991}
      minWidth={768}
    />,
    <Responsive
      as={[Function]}
      maxWidth={767}
      minWidth={320}
    />,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <Responsive
        as={[Function]}
        minWidth={992}
      />,
      <Responsive
        as={[Function]}
        maxWidth={991}
        minWidth={768}
      />,
      <Responsive
        as={[Function]}
        maxWidth={767}
        minWidth={320}
      />,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
